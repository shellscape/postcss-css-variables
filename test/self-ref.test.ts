import { fixture, process } from './helpers';

const tests = {
  'should work with circular reference': 'circular-reference',
  'should work with variables that try to self reference': 'self-reference',
  'should work with variables that try to self reference and fallback properly':
    'self-reference-fallback'
};

describe('self-ref', async () => {
  test.each(Object.entries(tests))('%s', async (_desc, file) => {
    const css = await fixture(`self-ref/${file}`);
    const result = await process(css);
    expect(result).toMatchSnapshot();
  });
});
