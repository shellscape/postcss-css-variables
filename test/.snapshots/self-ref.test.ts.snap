// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`self-ref > should work with circular reference 1`] = `"body{color:undefined}"`;

exports[`self-ref > should work with variables that try to self reference 1`] = `"body{color:undefined}"`;

exports[`self-ref > should work with variables that try to self reference and fallback properly 1`] = `"body{color:#0000ff;color:#00ff00}"`;
